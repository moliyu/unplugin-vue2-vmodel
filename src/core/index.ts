export const transform = (code: string) => {
  // const templateReg = /<template>[\S\s]*<\/template>/
  const vmodelReg = /(v-model:)(\S+)\s?=/g
  return code.replace(vmodelReg, (...args) => {
    const [_match, _vmodel, modelValue] = args
    if (modelValue === 'value') {
      return `v-model=`
    }
    return `:${modelValue}.sync=`
  })
}