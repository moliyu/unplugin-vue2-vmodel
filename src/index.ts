import { createUnplugin } from 'unplugin'
import type { Options } from './types'
import { transform } from './core'
export * from './core'

export default createUnplugin<Options>(options => ({
  name: 'unplugin-starter',
  transformInclude(id) {
    return id.endsWith('.vue')
  },
  transform(code) {
    return transform(code)
  },
}))
