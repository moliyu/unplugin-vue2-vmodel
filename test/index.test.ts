import { resolve } from 'path'
import { describe, expect, it } from 'vitest'
import fg from 'fast-glob'
import { promises as fs } from 'fs'
import {transform} from '../src'

describe('transform', () => {
  it('v-model', async() => {
    const root = resolve(__dirname)
    const files = fg.sync('vue/*.vue', {
      cwd: root,
      onlyFiles: true
    })
    for (let file of files) {
      const code = await fs.readFile(resolve(root, file), 'utf8')
      const ret = transform(code)
      expect(ret).toMatchSnapshot()
    }
  })
})
