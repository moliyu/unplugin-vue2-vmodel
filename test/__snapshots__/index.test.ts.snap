// Vitest Snapshot v1

exports[`transform > v-model 1`] = `
"<template>
  <div>
    <a-input v-model=\\"name\\"></a-input>
    <a-input :test.sync=\\"name\\"></a-input>
    <a-input :test1.sync=\\"name\\"></a-input>
  </div>
</template>

<script lang=\\"ts\\" setup>
const name = ref('')
</script>"
`;
